lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -225,7 +225,7 @@ sub _schedule_iso {
     my $skip_chained_deps  = delete $args->{_SKIP_CHAINED_DEPS} // 0;
 
     my $result = $self->_generate_jobs($args, \@notes, $skip_chained_deps);
-    return {error => $result->{error_message}} if defined $result->{error_message};
+    return {error => $result->{error_message}, error_code => $result->{error_code}} if defined $result->{error_message};
```

Comment:
```suggestion
    return {error => $result->{error_message}, error_code => $result->{error_code} // 400} if defined $result->{error_message};
```

Otherwise there wouldn't be an error code in the other error cases.